### PR TITLE
feat(fieldlabel)!: core-tokens implementation (CSS-102)

### PR DESCRIPTION
**Pull request by bernhard-adobe**
## Description
BREAKING CHANGE: This migrates the Field label component to core tokens.

https://jira.corp.adobe.com/browse/CSS-102
Web Component Implementation: https://github.com/adobe/spectrum-web-components/pull/2529 refactor(field-label): use core tokens #2529

## How and where has this been tested?
 - **How this was tested:** comparing http://localhost:3002/docs/fieldlabel.html with the https://opensource.adobe.com/spectrum-css/fieldlabel.html component before and after.
 - **Browser(s) and OS(s) this was tested with:** Chrome Canary Version 105.0.5148.0 (Official Build) canary (arm64)

## Screenshots
Screen Shot 2022-07-13 at 17 28 53

## To-do list

- [ ] If my change impacts other components, I have tested to make sure they don't break.
- [ ] If my change impacts documentation, I have updated the documentation accordingly.
- [x] I have read the [CONTRIBUTING document](/.github/CONTRIBUTING.md).
- [ ] I have tested these changes in Windows High Contrast mode.

- [ ] This pull request is ready to merge.

### CODE REVIEW

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [inline components/fieldlabel/index.css:67]: @pfulton some of those line-heights don't match exactly the previous line-heights.
Is this calc the correct way to get a line-height?

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [inline components/fieldlabel/index.css:104]: @pfulton  We are missing 20px from the core-tokens. Is it ok to get 20px with a core-tokens addition?

File: components/fieldlabel/skin.css
Comment (bernhard-adobe) [inline components/fieldlabel/skin.css:13]: @pfulton  should I use here core tokens colors? Or is the inheritance here ok?

File: components/tokens/custom-spectrum/custom-vars.css
Comment (bernhard-adobe) [inline components/tokens/custom-spectrum/custom-vars.css]: @pfulton  I am still not sure where to get those weights. I think it's still WIP for the core-tokens.

File: components/fieldlabel/index.css
Comment (pfulton) [reply components/fieldlabel/index.css:67]: Good question! I'll need to dig into this to get you a better answer.

File: components/fieldlabel/skin.css
Comment (pfulton) [reply components/fieldlabel/skin.css:13]: We should use the core tokens here. Also, we need to delete this `skin.css` file, and put the color stuff into the `index.css`

File: components/fieldlabel/index.css
Comment (pfulton) [reply components/fieldlabel/index.css:104]: Yeah, I think so. This will be something we'll need to keep an eye on, as I think I'd much rather have it coming as a definition from core tokens, but in the meantime (or if we have to), let's do this for now with an eye on possibly defining it ourselves in the `custom-*` files at some future point.

**Review by pfulton — CHANGES_REQUESTED** [review]
This is looking really great, @bernhard-adobe! Thanks so much for your work on this. I left a few comments here, and I think I answered the questions that you had left for me.

Two more things:
1. Let's be sure to add the `--mod-` prefixed custom properties in the places where we're using the other core tokens properties (look at ActionButton as your guide on this).
2. Let's be sure to add the `--highcontrast` prefixed custom properties in any places where we need high contrast mode to take effect.

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:28]: I noticed in the inspection of the current docs site that the line-height changes.
size S and M = 15.6px
size L = 18.2px
Size XL = 20.8px

Screen Shot 2022-07-18 at 15 33 00
Screen Shot 2022-07-18 at 15 33 11
Screen Shot 2022-07-18 at 15 33 16

I think from a visual perspective you do want to increase the line-heights for t-shirt sizing to keep the relational "growth" in sizing.

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:38]: Will add! Good feedback, thank you

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:48]: same, lets double-check, maybe in a design workshop. I think we will discuss soon again the core-tokens for font-naming and weight and this might be a good additional question.

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:67]: Sounds good, lets double check with design just to make sure this is intentional

Comment (lazd) [reply components/fieldlabel/index.css:67]: Yeah I'd avoid this `calc()` the values above should have the `calc`, if anything.

File: components/fieldlabel/index.css
Comment (lazd) [inline components/fieldlabel/index.css:137]: Any reason why this little fella isn't a component-level token above? It's not forbidden, per se, but I'd rather see it defined above for consistency.

Comment (bernhard-adobe) [reply components/fieldlabel/index.css:158]: @lazd I am actually not sure which item to use here. 
Spacing 100 would be 
```JSON
  "spacing-100": {
    "value": "8px"
  },
  ```
  
  Maybe use this? I don't see any other component level tokens for field-label matching 8px.

```JSON
  "field-label-top-to-asterisk-small": {
    "sets": {
      "desktop": {
        "value": "8px"
      },
      "mobile": {
        "value": "11px"
      }
    }
  },
  ```

File: components/fieldlabel/index.css
Comment (lazd) [reply components/fieldlabel/index.css]: That looks good to me! Is `CJK` as a prefix already an established convention? It seems odd to see capitalized letters, and I wonder if a suffix is more in line with what we have already. Your call.

File: components/fieldlabel/index.css
Comment (lazd) [reply components/fieldlabel/index.css:125]: If we're missing 20px in spacing values, then perhaps the component was designed wrong (or core tokens needs to have a new spacing value). We shouldn't be calcing this.

File: components/fieldlabel/index.css
Comment (lazd) [reply components/fieldlabel/index.css:158]: I think you just need to define that as a component-level token in CSS:

```
.spectrum--medium {
  --spectrum-field-label-top-to-asterisk-small: 8px;
}
.spectrum--large {
  --spectrum-field-label-top-to-asterisk-small: 11px;
}
```

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css]: > That looks good to me! Is `CJK` as a prefix already an established convention? It seems odd to see capitalized letters, and I wonder if a suffix is more in line with what we have already. Your call.

I think so - it's in the core tokens file. 
I think Jess would be the one that decides on the final token names.

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:32]: @pfulton I added the `--mod` modifiers. As there is no color-settings here I don't think we need to set `--highcontrast `

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css]: Added, thanks @lazd

File: components/fieldlabel/package.json
Comment (pfulton) [inline components/fieldlabel/package.json:20]: Let's bump this to the most recent release: 

```suggestion
    "@spectrum-css/tokens": "^1.0.4"
```

File: components/fieldlabel/package.json
Comment (pfulton) [inline components/fieldlabel/package.json:31]:
```suggestion
    "@spectrum-css/tokens": "^1.0.4",
```

File: components/fieldlabel/index.css
Comment (pfulton) [inline components/fieldlabel/index.css:134]: This is a place where we won't want to define the `--mod-` custom properties. Instead, we'd define them later when we actually use `--spectrum-tableform-margin-calc`. Does that make sense?

File: components/fieldlabel/index.css
Comment (pfulton) [inline components/fieldlabel/index.css:29]: I'm trying to catch as many of these as I can, and I know that the core tokens value _is_ capitalized, but I think we should standardize on making it lowercase when we create our own aliases.

```suggestion
  --spectrum-fieldlabel-cjk-line-height: var(--spectrum-CJK-line-height-100);
```

**Review by pfulton — CHANGES_REQUESTED** [review]
A few things here @bernhard-adobe - thanks so much for your hard work on this and your patience in waiting for my review.

**Comment by pfulton** [comment]
One more thing, @bernhard-adobe: the color for the `disabled` variant looks pretty different than it did before (see below with current and previous). Would you mind double-checking on that color?

disabled-fieldlabel

File: components/tokens/custom-spectrum/custom-vars.css
Comment (bernhard-adobe) [reply components/tokens/custom-spectrum/custom-vars.css]: Still waiting for the core-tokens but we have those for now temporarily here.

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css]: added to line 99

```CSS
/* international lang support */
.spectrum-FieldLabel {
  &:lang(ja),
  &:lang(zh),
  &:lang(ko) {
    line-height: var(--mod-fieldlabel-CJK-line-height, var(--spectrum-fieldlabel-CJK-line-height));
  }
}

File: components/fieldlabel/package.json
Comment (bernhard-adobe) [reply components/fieldlabel/package.json:20]: bumped up

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:134]: simplified!

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [reply components/fieldlabel/index.css:29]: made it as lower case as possible...

**Comment by bernhard-adobe** [comment]
> One more thing, @bernhard-adobe: the color for the `disabled` variant looks pretty different than it did before (see below with current and previous). Would you mind double-checking on that color?
> 
> disabled-fieldlabel

That is indeed missing as it was not documented in the "Components Batch 1.xd" file. I guess the color would be `(--spectrum-disabled-content-color)` like all the other disabled components.

**Comment by bernhard-adobe** [comment]
> > One more thing, @bernhard-adobe: the color for the `disabled` variant looks pretty different than it did before (see below with current and previous). Would you mind double-checking on that color?
> > disabled-fieldlabel
> 
> That is indeed missing as it was not documented in the "Components Batch 1.xd" file. I guess the color would be `(--spectrum-disabled-content-color)` like all the other disabled components.
Screen Shot 2022-08-11 at 16 17 12

**Comment by bernhard-adobe** [comment]
rebuild Jenkins: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/37/

VRT results are: https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/37/artifact/backstop_data/html_report/index.html

**Comment by bernhard-adobe** [comment]
@pfulton color changes (as expected) for disabled and line-height changes (as expected):
Screen Shot 2022-08-11 at 16 35 09

**Comment by pfulton** [comment]
Released beta:
`5.0.0-beta.0`

**Comment by pfulton** [comment]
New beta release:
`5.0.0-beta.1`

**Comment by pfulton** [comment]
New release with syntax fix:
`5.0.0-beta.2`

File: components/fieldlabel/index.css
Comment (bernhard-adobe) [inline components/fieldlabel/index.css:119]: This seems to be a problem for Web components, the padding might be too small.

**Comment by bernhard-adobe** [comment]
Fixed misaligned Field-Label padding for variant Side-Label.

@pfulton can you please publish another beta-release for the FieldLabel so I can pull in the latest changes in Web Component and re-run VRT over there? 

**VRT Results for Spectrum CSS are:** 
https://spectrum-visual-regression.ci.corp.adobe.com/view/Spectrum%20CSS/job/css-vrt-test/44/artifact/backstop_data/html_report/index.html

Line-height changed for FieldLabel. We have confirmed before that this is expected behavior.
With the latest push the Side-Label padding top  doesn't change anymore. This is in contrast with the documentation from the "Components batch 1.xd" file that says it should:
Screen Shot 2022-09-25 at 18 47 33

If this is an intended change we should roll-back my last commit:  044d8ad1f96ece84dadbbd1fb1877bfec7b18aca

Screen Shot 2022-09-25 at 18 27 13



Furthermore I aligned the CSS FieldLabel with the Web Components FieldLabel and the Line-Heights now match the Web Components. 

Because we changed the line-heights from previously 15.6 px to now 18px (no more sub-pixels) the position of the Label changes a bit. 
We may want to check in with design if this is intend or if we need to adjust the top-padding for the Side-Label variant.

 
Screen Shot 2022-09-25 at 18 43 42

Screen Shot 2022-09-25 at 18 42 06
Screen Shot 2022-09-25 at 18 38 35

Please let me know @pfulton  if this sounds good. Thank you. 


(there are VRT changes with Switch and Menu reported as well, I think the baseline-images haven not been updates)

**Comment by pfulton** [comment]
Released: `@spectrum-css/fieldlabel@5.0.0-beta.3`

**Pull request merged**
